### ISSUE TITLE
Allow to customize Google Drive client id

### ISSUE THREAD

**Issue by ManuCart** [keeweb/keeweb#180]
How to change the google drive ````client-secrets.json```` file for a personnal integration?

**Comment by antelle** [comment]
What is it?

**Comment by ManuCart** [comment]
This is the API to connect to google drive
https://developers.google.com/api-client-library/python/guide/aaa_client_secrets

**Comment by antelle** [comment]
Currently it's not possible to customize these settings.

**Comment by ManuCart** [comment]
Ok thanks, so at this time everyone use your Oauth2 profile ?

**Comment by antelle** [comment]
It's using pre-created app id for this, without a client secret. It's not possible to customize app id.

**Comment by ManuCart** [comment]
When I host the keeweb file on dropbox and I try to use gdrive to open kdbx files I have this error
````
400. That’s an error.

Error: redirect_uri_mismatch

The redirect URI in the request, https://dl.dropboxusercontent.com/u/xxxxxx/keeweb.html, does not match the ones authorized for the OAuth client. Visit https://console.developers.google.com/apis/credentials/oauthclient/847548101761-koqkji474gp3i2gn3k5omipbfju7pbt1.apps.googleusercontent.com?project=847548101761 to update the authorized redirect URIs.

Learn more

That’s all we know.
````
Impossible to change redirect URI because yours

**Comment by antelle** [comment]
This is mentioned in #179, will be fixed soon. Currently save to gdrive, dropbox, ... - works only for new files, not already saved.

**Comment by antelle** [comment]
Upcoming hotfix (v1.1.3) will contain a method for setting all app settings, including google drive client id:
```javascript
kwSettings.set('gdriveClientId', 'your-key-here')
```
So, you won't need to replace anything in the HTML file.
These settings will be stored in localStorage, which means for browser session.

**Comment by ManuCart** [comment]
Cool for this very important things.
Good job, nice app!

**Comment by antelle** [comment]
Regarding client-secrets.json: which settings could be customized here?
Does anybody need changing `auth_uri`?

**Comment by ManuCart** [comment]
For me, I never change ````auth_uri```` and ````token_uri````

**Comment by antelle** [comment]
I assume, they are endpoints which are never changed at all. So, all settings can be changed for now (except the GUI for that is missing for a while), because the app is working with a key-scheme, without storing a secret.

**Comment by antelle** [comment]
the feature was implemented in v1.2
